library/std/src/sys/wasm/alloc.rs

```diff
@@ -24,26 +24,30 @@ static mut DLMALLOC: dlmalloc::Dlmalloc = dlmalloc::DLMALLOC_INIT;
 unsafe impl GlobalAlloc for System {
     #[inline]
     unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        // SAFETY:  DLMALLOC.malloc() is guranteed to be safe since lock::lock() aqcuire a globl lock
         let _lock = lock::lock();
-        DLMALLOC.malloc(layout.size(), layout.align())
+        unsafe { DLMALLOC.malloc(layout.size(), layout.align()) }
```

Comment:
There are two preconditions that we need to document for these calls:

 * DLMALLOC access is safe: this is true because the lock gives us unique and non-reentrant access.
 * Calling the function itself is safe: Preconditions on these functions match the trait method preconditions, so this is safe.

The comments so far only sort of talk about the first, but should address both. I think basically copy/pasting my two bullet points makes sense.